### ISSUE TITLE
Jekyll 3.0 support

### ISSUE THREAD

**Issue by eirc** [slashdotdash/jekyll-lunr-js-search#86]
Looks like the plugin does not work with the new Jekyll 3.0 that was recently released.

I'm no expert on Jekyll or this plugin so I cannot help a lot with more info apart from the fact that the first thing that fails is a reference to `Jekyll::Post` which no longer seems to exist.

**Comment by eirc** [comment]
About the missing `Jekyll::Post` error: a [similar error](https://talk.jekyllrb.com/t/where-is-jekyll-post-in-jekyll-3/1441) reported in the Jekyll Talk page was directed to be implemented with the use of [hooks](http://jekyllrb.com/docs/plugins/#hooks).

**Comment by kashyapc** [comment]
I can confirm that using the `3.0.0.pre.beta10` version fixes an issue I was seeing:

    /home/kashyapc/.gem/ruby/gems/jekyll-lunr-js-search-0.3.0/lib/jekyll_lunr_js_search/search_entry.rb:8:in `create': cannot load such file -- jekyll/post (LoadError)

The above is fixed with `3.0.0.pre.beta10` version.
    gem uninstall jekyll && gem install jekyll --version 3.0.0.pre.beta10

Thanks to Josef Stribny for pointing this out.

**Comment by moneypenny** [comment]
Here's the output I see on version 0.3.0 of the gem:

```
      Generating...
              Lunr: Creating search index...
       Deprecation: Collection#each should be called on the #docs array directly.
                    Called by /Users/me/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/jekyll-lunr-js-search-0.3.0/lib/jekyll_lunr_js_search/indexer.rb:133:in `pages_to_index'.
/Users/me/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/jekyll-lunr-js-search-0.3.0/lib/jekyll_lunr_js_search/search_entry.rb:8:in `create': cannot load such file -- jekyll/post (LoadError)
  from /Users/me/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/jekyll-lunr-js-search-0.3.0/lib/jekyll_lunr_js_search/indexer.rb:64:in `block in generate'
```

**Comment by moneypenny** [comment]
With version 0.3.0 of jekyll-lunr-js-search and 3.0.0.pre.beta10 of jekyll, I get this error:

```
Lunr: Index ready (lunr.js v0.4.5)
/Users/me/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/jekyll-3.0.0.pre.beta10/lib/jekyll/cleaner.rb:75:in `parent_dirs': stack level too deep (SystemStackError)
  from /Users/me/.rbenv/versions/2.2.2/lib/ruby/gems/2.2.0/gems/jekyll-3.0.0.pre.beta10/lib/jekyll/cleaner.rb:75:in `parent_dirs'
```

**Comment by slashdotdash** [comment]
There's an open Pull Request (#85) to add support for Jekyll 3. It just needs to be verified, merged and published to Ruby Gems.

**Comment by slashdotdash** [comment]
I've just pushed an update to support Jekyll 3 and have published v3.0.0 of the [jekyll-lunr-js-search](https://rubygems.org/gems/jekyll-lunr-js-search/versions/3.0.0) gem.

Install the gem as follows:

    gem install jekyll-lunr-js-search -v3.0.0
